micro-rdk/src/esp32/encoder.rs

```diff
@@ -57,6 +57,32 @@ pub struct Esp32Encoder<A, B> {
     b: B,
 }
 
+pub struct Esp32EncoderConfig {
+    pub(crate) a: i32,
+    pub(crate) b: i32,
+}
+
+impl TryFrom<&Kind> for Esp32EncoderConfig {
+    type Error = AttributeError;
+    fn try_from(value: &Kind) -> Result<Self, Self::Error> {
+        let a = value.get("a")?;
+        let a = a
+            .map(|v| v.try_into())
+            .ok_or(AttributeError::ValidationError(format!(
+                "`{:?}` is an invalid value for pin `a",
+                a
+            )))??;
+        let b = value.get("b")?;
+        let b = b
+            .map(|v| v.try_into())
+            .ok_or(AttributeError::ValidationError(format!(
+                "`{:?}` is an invalid value for pin `b`",
+                b
+            )))??;
```

Comment:
just a question when using ?? doesn't that potentially silence the intermediate error if two are chained?

Reply:
I missed this first time through: this is happening because of the call to `ok_or` on an `Option`, which ends up creating a `Result<Result`

Reply:
Note: while verifying output for different approaches to this, found that we don't actually propagate useful, descriptive errors to the user. made a [ticket](https://viam.atlassian.net/browse/RSDK-10534?atlOrigin=eyJpIjoiODdmNjk2YjhkMDg1NDNhMTgyYjlkYjhlMjVmNDNlM2MiLCJwIjoiaiJ9)